### PR TITLE
#250 Allow case insensitive property validation

### PR DESCRIPTION
**Pull request by asleire**
Added a `JsonSchemaValidatorOptions` class with a `IgnorePropertyNameCase` property as per your suggestion in #250

### CODE REVIEW

**Comment by asleire** [comment]
Closes #1267

**Comment by seungyongshim** [comment]
+1

File: src/NJsonSchema/Validation/JsonSchemaValidatorOptions.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace NJsonSchema.Validation
+{
+    /// <summary>Class to configure the behavior of <see cref="JsonSchemaValidator"/>. </summary>
+    public class JsonSchemaValidatorOptions
+    {
+        /// <summary>Whether to ignore casing for object properties.</summary>
+        public bool IgnorePropertyNameCase { get; set; }
```
Comment (lahma) [inline src/NJsonSchema/Validation/JsonSchemaValidatorOptions.cs:9]: would it make sense to make this `public StringComparer PropertyNameComparer`?

File: src/NJsonSchema/Validation/JsonSchemaValidatorOptions.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace NJsonSchema.Validation
+{
+    /// <summary>Class to configure the behavior of <see cref="JsonSchemaValidator"/>. </summary>
+    public class JsonSchemaValidatorOptions
```
Comment (RicoSuter) [inline src/NJsonSchema/Validation/JsonSchemaValidatorOptions.cs:6]: To be consistent, we should rename this to *Settings.

File: src/NJsonSchema/Validation/JsonSchemaValidatorOptions.cs (hunk 1 above)
Comment (asleire) [reply src/NJsonSchema/Validation/JsonSchemaValidatorOptions.cs:9]: I don't mind either way. A `StringComparer` is obviously more flexible. @RicoSuter ?

Comment (RicoSuter) [reply src/NJsonSchema/Validation/JsonSchemaValidatorOptions.cs:9]: Fine for me either way.. as long it's an option.
To me it does not make much sense to allow case insensitive validation - it's probably against JSON Schema rules :-)

Comment (lahma) [reply src/NJsonSchema/Validation/JsonSchemaValidatorOptions.cs:9]: StringComparer has the advantage that you can write a comparer like `FirstCharIgnoringComparer` which is convenient in some interop scenarios between JSON and C# classes - not sure if it applies here.

Comment (asleire) [reply src/NJsonSchema/Validation/JsonSchemaValidatorOptions.cs]: I updated the PR with `StringComparer`

To me, I've always used case insensitive deserialization from JSON -> C# classes for convenience. Now I have a scenario where I want to use JsonSchema to validate certain JSON structures, and I want to preserve the case insensitivity for consistency

**Comment by glecam** [comment]
First time using this library and it's great. I too would like to have this feature introduced for case insensitive deserialization from JSON -> C# classes.

**Comment by RicoSuter** [comment]
Thanks for the PR.

**Comment by asleire** [comment]
@Anit11 

Example usage:
 ```csharp
var jsonSchemaValidator = new JsonSchemaValidator(new JsonSchemaValidatorOptions()
{
    IgnorePropertyNameCase = true,
});

string json = ...; // Get this from somewehre
string jsonSchema = ...; // Get this from somewhere

var validationErrors = jsonSchemaValidator.Validate(json, jsonSchema);
```

> I want to use it for enum array property

Note that this PR is for property names, not property values. If you want case insensitive enum values that's not covered by this PR

**Comment by Anit11** [comment]
Thanks for the example. I see, I actually want to validate the property value which is an enum array (case-insensitively). Is that supported? Below is the example and would like to validate Enum color (RED, BLUE)
public class ColorTypes
{
    public Color[] Fill { get; set; }
}
public enum Color {
    RED,
    BLUE
}
JsonSchema ColorSchema = JsonSchema.FromType<ColorTypes>();
ICollection<ValidationError> validationResult = ColorSchema .Validate(jsonObject);

**Pull request merged**